tesseract_scene_graph/include/tesseract_scene_graph/graph.h

```diff
@@ -130,7 +130,9 @@ struct ShortestPath
 
 class SceneGraph
 #ifndef SWIG
-  : public Graph
+  : public Graph  //, public boost::noncopyable This data structure
+// should be declared noncopyable, but serialization currently
+// is creating copies
```

Comment:
Should this be removed?

Reply:
The structure should be declared `boost::noncopyable` at some point, but not in this PR. An issue should be created to fix this at some point.

Reply:
Yea, it is odd that the copy destructor is deleted but on windows it is still coping? I see where you were switched to using out parameter but those should have been moving the object and not copying it correct?

Reply:
Is its that we are using windows compiler 2019 which may not have a full implementation of c++17 standard so it is not correctly moving the object. Is it possible to upgrade to 2020? Also did you by chance try adding an explicit std::move before switching to using an out parameter?

Reply:
I tried adding `boost::noncopyable` to `SceneGraph` on Linux and am seeing the same compiler errors:

```
tesseract_scene_graph_serialization_unit.cpp:212:10: error: use of deleted function ‘tesseract_scene_graph::SceneGraph::SceneGraph(const tesseract_scene_graph::SceneGraph&)’
  212 |   return g;
```

I think you still need to have the object be copyable to use copy elision, since it may still copy in certain situations.

Reply:
I tried defining a dummy move constructor after adding `boost::noncopyable`, and it compiles. It looks like the default move constructor isn't being generated correctly.

Reply:
It compiles using GCC, I still have to test MSVC with an explicit move constructor.

Reply:
No, I am seeing the same errors with an explicit move constructor that I was seeing earlier. I think SceneGraph needs to be allocated on the heap, or used as a reference.

Reply:
Then should we just add the boost::noncopyable or does that cause issues other places?

Reply:
I tried adding rebuilding `link_map_` and `joint_map_` to the move constructor, added `boost::noncopyable`, and it passed the scene graph unit tests. 

```
SceneGraph(SceneGraph&& other) : Graph(std::forward<Graph>(other)), link_map_(std::move(other.link_map_)),
    joint_map_(std::move(other.joint_map_)), acm_(std::move(other.acm_))
  {
    {
      link_map_.clear();
      joint_map_.clear();
      Graph::vertex_iterator i, iend;
      for (boost::tie(i, iend) = boost::vertices(*this); i != iend; ++i)
      {
        Link::Ptr link = boost::get(boost::vertex_link, *this)[*i];
        link_map_[link->getName()] = std::make_pair(link, *i);
      }
    }
    {
      Graph::edge_iterator i, iend;
      for (boost::tie(i, iend) = boost::edges(*this); i != iend; ++i)
      {
        Joint::Ptr joint = boost::get(boost::edge_joint, *this)[*i];
        joint_map_[joint->getName()] = std::make_pair(joint, *i);
      }
    }
  }
```